src/main/java/io/deepstream/DeepstreamClient.java

```diff
@@ -239,7 +240,7 @@ public void loginFailed(Event errorEvent, Object data) {
         });
 
         try {
-            loggedInLatch.await();
+            loggedInLatch.await(3, TimeUnit.SECONDS);
```

Comment:
Add a timeout so that this does not hang the application. 3sec chosen arbitrarily to keep this change to the SDK to a min. App will still retry in a loop but each attempt will last at most 3sec.